src/renderer/dx/DxRenderer.cpp

```diff
@@ -2044,22 +2020,8 @@ float DxEngine::GetScaling() const noexcept
                                                 FontInfo& pfiFontInfo,
                                                 int const iDpi) noexcept
 {
-    Microsoft::WRL::ComPtr<IDWriteTextFormat> format;
-    Microsoft::WRL::ComPtr<IDWriteTextFormat> formatItalic;
-    Microsoft::WRL::ComPtr<IDWriteTextAnalyzer1> analyzer;
-    Microsoft::WRL::ComPtr<IDWriteFontFace1> face;
-    Microsoft::WRL::ComPtr<IDWriteFontFace1> faceItalic;
-    LineMetrics lineMetrics;
-
-    return _GetProposedFont(pfiFontInfoDesired,
-                            pfiFontInfo,
-                            iDpi,
-                            format,
-                            formatItalic,
-                            analyzer,
-                            face,
-                            faceItalic,
-                            lineMetrics);
+    DxFontRenderData fontRenderData(_dwriteFactory);

```

Comment:
A new instance is used. I think this is the correct way, but I'm not entirely sure.

Reply:
A new instance? Or is it just doing an AddRef on it because it's a ComPtr? The latter is OK.

Reply:
The old `_GetProposedFont` is stateless . But the `_fontRenderData` has the state of the currently used fonts. I don't think  `GetProposedFont` should change the inner state of the renderer, right? It's supposed to only resolve the `pfiFontInfo`.

Reply:
This seems correct. It used to use _GetProposedFont with completely empty COM pointers, and now it's doing the same thing using a DxFontRenderData instance.